blaze-client/src/main/scala/org/http4s/client/blaze/PooledHttp1Client.scala

```diff
@@ -11,14 +11,20 @@ object PooledHttp1Client {
   /** Construct a new PooledHttp1Client
     *
     * @param maxTotalConnections maximum connections the client will have at any specific time
+    * @param maxConnectionsPerRequestKey Map of RequestKey to number of max connections
     * @param config blaze client configuration options
     */
   def apply[F[_]: Effect](
       maxTotalConnections: Int = DefaultMaxTotalConnections,
+      maxConnectionsPerRequestKey: Map[RequestKey, Int] = Map.empty,
```

Comment:
Do you still prefer this to a `RequestKey => Int`?  I think I'm on Team Function.

Reply:
That will preclude people from being able to supply partial maps, which would be inconvenient in my opinion.